### ISSUE TITLE
:tractor: Migrate to django-fsm-2

### ISSUE THREAD

**Issue by jefftriplett** [jazzband/django-fsm-log#188]
I recently switched to https://pypi.org/project/django-fsm-2/, and dropping the `django_fsm` setup requirement would still allow other spiritual successor projects to work. 
https://github.com/jazzband/django-fsm-log/blob/master/setup.py#L22 

I'm happy to create a new issue and submit a PR if this is helpful?

**Comment by lorenzomorandini** [comment]
Fine for me, since django-fsm (1) won't be updated anymore I don't see a reason to keep it as dependency. Still we should consider to document this, especially if django-fsm-2 will ever introduce breaking changes with django-fsm.

**Comment by pfouque** [comment]
>  especially if django-fsm-2 will ever introduce breaking changes with django-fsm.

That's not the plan! ;)
The main idea is to keep publishing releases according to Python & Django version upgrades.
I may add type hints at some point, but feature-wise, the scope is complete IMO!

**Comment by ticosax** [comment]
Fixed by https://github.com/jazzband/django-fsm-log/pull/192
I will try to make a release shortly.

**Comment by wgordon17** [comment]
Looking forward to the updated release 🙌

**Comment by ticosax** [comment]
Release job failed.
https://github.com/jazzband/django-fsm-log/actions/runs/12650627360/job/35249516513

Not sure what are the next steps.

**Comment by ticosax** [comment]
still failing https://github.com/jazzband/django-fsm-log/actions/runs/12650915962/job/35250402132
:thinking:

**Comment by MRigal** [comment]
The Received "500: INTERNAL SERVER ERROR" comes from the jazzband website... 
```
Uploading distributions to https://jazzband.co/projects/django-fsm-log/upload
Uploading django_fsm_log-4.0.1-py3-none-any.whl
WARNING  Received "500: INTERNAL SERVER ERROR"
```

@jezdez any hint?

**Comment by Rebely22** [comment]
I see new version with fsm2 support already uploaded , shall we close this issue?

**Comment by ticosax** [comment]
> I see new version with fsm2 support already uploaded , shall we close this issue?

It's not on [pypi](https://pypi.org/project/django-fsm-log/), curious to know where did you see it ?

**Comment by Rebely22** [comment]
> > I see new version with fsm2 support already uploaded , shall we close this issue?
> 
> It's not on [pypi](https://pypi.org/project/django-fsm-log/), curious to know where did you see it ?

No, it's not on PyPI. My apologies.

**Comment by lorenzomorandini** [comment]
@ticosax should we trigger a new pipeline? Maybe it's been fixed

**Comment by ticosax** [comment]
same result https://github.com/jazzband/django-fsm-log/actions/runs/13851917439/job/38760551908
I have slightly hope switching to uv could fix it 🤷‍♂ 
https://github.com/jazzband/django-fsm-log/pull/212
But I  can't do it alone. I can't merge unless it is approved. same for some other chores pull requests.

**Comment by lorenzomorandini** [comment]
Right now #212 has merge conflicts but once fixed I can approve it

**Comment by ticosax** [comment]
I was waiting for a signal to do, it. Thanks

**Comment by ticosax** [comment]
😢 https://github.com/jazzband/django-fsm-log/actions/runs/13852957773/job/38763559582

**Comment by LeeTomahawk** [comment]
Any progress?

**Comment by dwasyl** [comment]
Any idea what the blockers are here?

**Comment by ticosax** [comment]
> Any idea what the blockers are here?

https://github.com/jazzband/help/issues/386

**Comment by pfouque** [comment]
FYI, a pretty new alternative organization has been created with a similar goal than jazzband, but with a solid team of "roadies":
https://github.com/django-commons

Maybe they can help you transfer the project (they already did). We will also need a Jazzband roadie, but only once.

NB: I transferred django-fsm-2 and it was smooth!

**Comment by ticosax** [comment]
> FYI, a pretty new alternative organization has been created with a similar goal than jazzband, but with a solid team of "roadies": https://github.com/django-commons
> 
> Maybe they can help you transfer the project (they already did). We will also need a Jazzband roadie, but only once.
> 
> NB: I transferred django-fsm-2 and it was smooth!

Might be the time, indeed. I'm also in favor also of requesting a transfer to django-commons
@MRigal @lorenzomorandini are you willing to join django-commons and continue to maintain the project over there ?

**Comment by lorenzomorandini** [comment]
Totally up for it yes, seems time indeed.

**Comment by pfouque** [comment]
Alternatively, we could also integrate a logging solution (as an opt-in contrib) into the main project.

As I was investigating to integrate an admin mixin, I noticed a dependency in the django-fsm-admin to make it compatible with this library.

Having everything in a single repository/package would be easier to maintain and setup for the users

**Comment by lorenzomorandini** [comment]
That’s fine for me too, should be quite easy to integrate

**Comment by ticosax** [comment]
> Alternatively, we could also integrate a logging solution (as an opt-in contrib) into the main project.
> 
> As I was investigating to integrate an admin mixin, I noticed a dependency in the django-fsm-admin to make it compatible with this library.
> 
> Having everything in a single repository/package would be easier to maintain and setup for the users

The idea is appealing and I like it very much. I foresee a considerable challenge towards preserving the existing data though.
we could even take this opportunity to adopt the abstract model as materialized in https://github.com/jazzband/django-fsm-log/pull/176. But that should be discussed separately.

**Comment by pfouque** [comment]
> we could even take this opportunity to adopt the abstract model as materialized in https://github.com/jazzband/django-fsm-log/pull/176.
It's exactly what I had in mind! ;) 

Let's discuss that in another thread: https://github.com/django-commons/django-fsm-2/issues/47

**Comment by aaron-sao** [comment]
Is this just stuck now? Or is there an fsm-2 compatible version now someplace?

**Comment by ticosax** [comment]
https://pypi.org/manage/project/django-fsm-log/release/5.0.2/

**Comment by specialorange** [comment]
I finally got RX up and running with all three (admin, logs, and fsm) as a drop-in back to Django 3 and manageable steps to Django 2

https://github.com/specialorange/django-fsm-rx

It tries to address all the issues from admin, logging, FSM, and additional upgrades. Try it and let me know there

**Comment by ticosax** [comment]
@specialorange I haven't tried it yet, but It looks impressive at first glance. With the effort you put in facilitating the migration, it must have been a tremendous task.
Looks like you got it right, thanks for sharing this.

Now, I wonder what should become of the existing projects ? we might need some time to decide if we want to wind them down or not, in favor of django-fsm-rx
Also, which collaborative model you'd like to have for django-fsm-rx ? Are you willing to take the maintenance burden on your own ?

**Comment by specialorange** [comment]
A couple of things - I did try the community approach, but my understanding of submitting GitHub pulls with individual commits is conflicting with others' understanding.   I'm a single developer (Actually, I turn wrenches for a diesel repair company), so I'm just trying to make our lives easier with job tracking, and I don't know other use cases.   It's double-edged.    If it needs fixing, I fix it, just don't want to hinder others, and I may not understand others' needs/approaches exactly.    The hardest part for me was getting all the other tests, envs, and infrastructure set up.  

I am willing to take on the burden, as I don't think it would be much, as the rsx is now tied to my project (it's been in prod for 8 years, and I have had to update over the years).   I don't know any other collaborative models you appear to be referencing.  Move it to the django-commons/jazzband and I am a member?

To help you all understand my approaches: 
 - documentation and examples throughout
 - progression over argument
 - "simple" fixes should get done.  "simple" fixes can need "simple" explanations when they are not simple.
 - Timelines should be provided if against the grain, more challenging, or problematic.
 - always willing to learn and be persuaded

I don't know how a user is supposed to share their issue other than the standard ways I've seen on GH, they share some errors and growing blobs trying to get their issue addressed.   Nothing I haven't seen and feel most people needing an issue keep the conversation progressing.


Thanks for the community :)
